### ISSUE TITLE
neuronx-cc 2.12.46: Access pattern out of bound

### ISSUE THREAD

**Issue by xanderdunn** [aws-neuron/aws-neuron-sdk#803]
We have three graphs that compiled successfully on `NeuronX Compiler version 2.12.40.0+4928b2c8b`, but on `NeuronX Compiler version 2.12.46.0+8a56b1c94` they do not compile.

Graphs:
- [rust_hlo_transformer_reproducibility_with_dropout.pb.zip](https://github.com/aws-neuron/aws-neuron-sdk/files/13651460/rust_hlo_transformer_reproducibility_with_dropout.pb.zip)
- [rust_hlo_transformer_xla_benchmark.pb.zip](https://github.com/aws-neuron/aws-neuron-sdk/files/13651469/rust_hlo_transformer_xla_benchmark.pb.zip)
- [rust_hlo_transformer_tree_attention.pb.zip](https://github.com/aws-neuron/aws-neuron-sdk/files/13651474/rust_hlo_transformer_tree_attention.pb.zip)

Example output on the first graph:
```
---- nn::transformer::tests::transformer_reproducibility stdout ----
Generating inputs...
Done generating inputs.
thread 'nn::transformer::tests::transformer_reproducibility' panicked at xla/src/xla_utils.rs:51:5:
Script '"neuronx-cc" "compile" "/mnt/drive1/tmp/rust_hlo_transformer_reproducibility_with_dropout.pb" "--framework=XLA" "--target=trn1" "--model-type=transformer" "--auto-cast=none" "--optlevel=1" "--output" "/mnt/drive1/tmp/transformer_reproducibility_with_dropout_12725341889216953645.neff"' failed:
2023-12-12T17:28:24Z === BIR error ===
Reason: Access pattern out of bound.
Instruction: I-2092-0-act
Opcode: Activation
Argument AP:
Access Pattern: [[6,64],[1,1],[1,6]]
Offset: 384
Memory Location: {_bwd_mlp_pad_140_hlo_id_140__mhlo.pad_33}@PSUM0<0,0>(64x24)#Internal DebugInfo: <_bwd_mlp_pad_140_hlo_id_140__mhlo.pad_33||UNDEF||[64, 6, 1]>
, stdout:
2023-12-12T17:28:02Z Compilation is optimized for best performance and compilation time. For faster compilation time please use -O1
2023-12-12T17:28:03Z Running DoNothing
2023-12-12T17:28:03Z DoNothing finished after 0.000 seconds
2023-12-12T17:28:03Z Running AliasDependencyInduction
2023-12-12T17:28:03Z AliasDependencyInduction finished after 0.002 seconds
2023-12-12T17:28:03Z Running CanonicalizeIR
2023-12-12T17:28:03Z CanonicalizeIR finished after 0.005 seconds
2023-12-12T17:28:03Z Running LegalizeCCOpLayout
2023-12-12T17:28:03Z LegalizeCCOpLayout finished after 0.002 seconds
2023-12-12T17:28:03Z Running ExpandBatchNorm
2023-12-12T17:28:04Z ExpandBatchNorm finished after 0.040 seconds
2023-12-12T17:28:04Z Running ResolveComplicatePredicates
2023-12-12T17:28:04Z ResolveComplicatePredicates finished after 0.004 seconds
2023-12-12T17:28:04Z Running AffinePredicateResolution
2023-12-12T17:28:04Z AffinePredicateResolution finished after 0.003 seconds
2023-12-12T17:28:04Z Running EliminateDivs
2023-12-12T17:28:04Z EliminateDivs finished after 0.005 seconds
2023-12-12T17:28:04Z Running PerfectLoopNest
2023-12-12T17:28:04Z PerfectLoopNest finished after 0.003 seconds
2023-12-12T17:28:04Z Running Simplifier
2023-12-12T17:28:04Z Simplifier finished after 0.038 seconds
2023-12-12T17:28:04Z Running GenericAccessSimplifier
2023-12-12T17:28:04Z GenericAccessSimplifier finished after 0.002 seconds
2023-12-12T17:28:04Z Running TCTransform
2023-12-12T17:28:04Z TCTransform finished after 0.018 seconds
2023-12-12T17:28:04Z Running CommuteConcat
2023-12-12T17:28:04Z CommuteConcat finished after 0.002 seconds
2023-12-12T17:28:04Z Running LowerTensorOp
2023-12-12T17:28:04Z LowerTensorOp finished after 0.021 seconds
2023-12-12T17:28:04Z Running TCTransform
2023-12-12T17:28:05Z TCTransform finished after 0.016 seconds
2023-12-12T17:28:05Z Running CanonicalizeIR
2023-12-12T17:28:05Z CanonicalizeIR finished after 0.016 seconds
2023-12-12T17:28:05Z Running TensorOpFusion
2023-12-12T17:28:05Z TensorOpFusion finished after 0.009 seconds
2023-12-12T17:28:05Z Running TensorOpTransform
2023-12-12T17:28:05Z TensorOpTransform finished after 0.024 seconds
2023-12-12T17:28:05Z Running LateLowerTensorOp
2023-12-12T17:28:05Z LateLowerTensorOp finished after 0.018 seconds
2023-12-12T17:28:05Z Running MemcpyElimination
2023-12-12T17:28:05Z MemcpyElimination finished after 0.108 seconds
2023-12-12T17:28:05Z Running LoopFusion
2023-12-12T17:28:05Z LoopFusion finished after 0.133 seconds
2023-12-12T17:28:05Z Running Simplifier
2023-12-12T17:28:06Z Simplifier finished after 0.018 seconds
2023-12-12T17:28:06Z Running Delinearization
2023-12-12T17:28:06Z Delinearization finished after 0.033 seconds
2023-12-12T17:28:06Z Running AliasDependencyElimination
2023-12-12T17:28:06Z AliasDependencyElimination finished after 0.016 seconds
2023-12-12T17:28:06Z Running DeadStoreElimination
2023-12-12T17:28:06Z DeadStoreElimination finished after 0.197 seconds
2023-12-12T17:28:06Z Running AliasDependencyInduction
2023-12-12T17:28:06Z AliasDependencyInduction finished after 0.001 seconds
2023-12-12T17:28:06Z Running Simplifier
2023-12-12T17:28:06Z Simplifier finished after 0.017 seconds
2023-12-12T17:28:06Z Running LICM
2023-12-12T17:28:07Z LICM finished after 0.011 seconds
2023-12-12T17:28:07Z Running Delinearization
2023-12-12T17:28:07Z Delinearization finished after 0.015 seconds
2023-12-12T17:28:07Z Running LoopFusion
2023-12-12T17:28:07Z LoopFusion finished after 0.056 seconds
2023-12-12T17:28:07Z Running SimplifySlice
2023-12-12T17:28:07Z SimplifySlice finished after 0.012 seconds
2023-12-12T17:28:07Z Running LICM
2023-12-12T17:28:07Z LICM finished after 0.016 seconds
2023-12-12T17:28:07Z Running Simplifier
2023-12-12T17:28:07Z Simplifier finished after 0.017 seconds
2023-12-12T17:28:07Z Running ValueNumbering
2023-12-12T17:28:07Z ValueNumbering finished after 0.013 seconds
2023-12-12T17:28:07Z Running LICM
2023-12-12T17:28:07Z LICM finished after 0.017 seconds
2023-12-12T17:28:07Z Running PadElimination
2023-12-12T17:28:07Z PadElimination finished after 0.001 seconds
2023-12-12T17:28:07Z Running Delinearization
2023-12-12T17:28:08Z Delinearization finished after 0.006 seconds
2023-12-12T17:28:08Z Running LoopFusion
2023-12-12T17:28:08Z LoopFusion finished after 0.034 seconds
2023-12-12T17:28:08Z Running GenericAccessSimplifier
2023-12-12T17:28:08Z GenericAccessSimplifier finished after 0.002 seconds
2023-12-12T17:28:08Z Running Simplifier
2023-12-12T17:28:08Z Simplifier finished after 0.006 seconds
2023-12-12T17:28:08Z Running LICM
2023-12-12T17:28:08Z LICM finished after 0.004 seconds
2023-12-12T17:28:08Z Running ValueNumbering
2023-12-12T17:28:08Z ValueNumbering finished after 0.024 seconds
2023-12-12T17:28:08Z Running TCTransform
2023-12-12T17:28:08Z TCTransform finished after 0.019 seconds
2023-12-12T17:28:08Z Running CommuteConcat
2023-12-12T17:28:08Z CommuteConcat finished after 0.002 seconds
2023-12-12T17:28:08Z Running RecognizeOpIdiom
2023-12-12T17:28:08Z RecognizeOpIdiom finished after 0.012 seconds
2023-12-12T17:28:08Z Running MaskPropagation
2023-12-12T17:28:09Z MaskPropagation finished after 0.022 seconds
2023-12-12T17:28:09Z Running DeadStoreElimination
2023-12-12T17:28:09Z DeadStoreElimination finished after 0.140 seconds
2023-12-12T17:28:09Z Running Recompute
2023-12-12T17:28:09Z Recompute finished after 0.001 seconds
2023-12-12T17:28:09Z Running DeadCodeElimination
2023-12-12T17:28:09Z DeadCodeElimination finished after 0.003 seconds
2023-12-12T17:28:09Z Running DoNothing
2023-12-12T17:28:09Z DoNothing finished after 0.000 seconds
2023-12-12T17:28:09Z Running MutateDataType
2023-12-12T17:28:09Z MutateDataType finished after 0.012 seconds
2023-12-12T17:28:09Z Running AutoCastTCInputs
2023-12-12T17:28:10Z AutoCastTCInputs finished after 0.012 seconds
2023-12-12T17:28:10Z Running GenericAccessSimplifier
2023-12-12T17:28:10Z GenericAccessSimplifier finished after 0.013 seconds
2023-12-12T17:28:10Z Running Simplifier
2023-12-12T17:28:10Z Simplifier finished after 0.016 seconds
2023-12-12T17:28:10Z Running AliasDependencyElimination
2023-12-12T17:28:10Z AliasDependencyElimination finished after 0.012 seconds
2023-12-12T17:28:10Z Running DelinearIndices
2023-12-12T17:28:10Z DelinearIndices finished after 0.015 seconds
2023-12-12T17:28:10Z Running Delinearization
2023-12-12T17:28:10Z Delinearization finished after 0.018 seconds
2023-12-12T17:28:10Z Running DelinearIndices
2023-12-12T17:28:10Z DelinearIndices finished after 0.012 seconds
2023-12-12T17:28:10Z Running DeadCodeElimination
2023-12-12T17:28:10Z DeadCodeElimination finished after 0.003 seconds
2023-12-12T17:28:10Z Running InferIntrinsicOnCC
2023-12-12T17:28:10Z InferIntrinsicOnCC finished after 0.035 seconds
2023-12-12T17:28:10Z Running ResolveAccessConflict
2023-12-12T17:28:11Z ResolveAccessConflict finished after 0.027 seconds
2023-12-12T17:28:11Z Running LICM
2023-12-12T17:28:11Z LICM finished after 0.017 seconds
2023-12-12T17:28:11Z Running LocalLayoutOpt
2023-12-12T17:28:11Z LocalLayoutOpt finished after 0.021 seconds
2023-12-12T17:28:11Z Running DelinearIndices
2023-12-12T17:28:11Z DelinearIndices finished after 0.020 seconds
2023-12-12T17:28:11Z Running PGLayoutTilingPipeline
2023-12-12T17:28:11Z Running PAGLayoutOpt
2023-12-12T17:28:11Z Running Delinearization
2023-12-12T17:28:11Z Delinearization finished after 0.013 seconds
2023-12-12T17:28:11Z PAGLayoutOpt finished after 0.199 seconds
2023-12-12T17:28:11Z Running MaskPropagation
2023-12-12T17:28:12Z MaskPropagation finished after 0.009 seconds
2023-12-12T17:28:12Z Running CanonicalizeDAGForPGTiling
2023-12-12T17:28:12Z CanonicalizeDAGForPGTiling finished after 0.021 seconds
2023-12-12T17:28:12Z Running PGTiling
2023-12-12T17:28:12Z Running AGOrderingAnalysisPass
2023-12-12T17:28:12Z AGOrderingAnalysisPass finished after 0.038 seconds
2023-12-12T17:28:12Z Running CuttingAndMacroGeneration
2023-12-12T17:28:12Z CuttingAndMacroGeneration finished after 0.076 seconds
2023-12-12T17:28:12Z PGTiling finished after 0.379 seconds
2023-12-12T17:28:12Z Running InsertIOTransposes
2023-12-12T17:28:12Z InsertIOTransposes finished after 0.035 seconds
2023-12-12T17:28:13Z PGLayoutTilingPipeline finished after 1.376 seconds
2023-12-12T17:28:13Z Running TilingProfiler
2023-12-12T17:28:13Z TilingProfiler finished after 0.024 seconds
2023-12-12T17:28:13Z Running FlattenMacroLoop
2023-12-12T17:28:13Z FlattenMacroLoop finished after 0.025 seconds
2023-12-12T17:28:13Z Running InferTongaTensor
2023-12-12T17:28:13Z InferTongaTensor finished after 0.073 seconds
2023-12-12T17:28:13Z Running TongaSimplifier
2023-12-12T17:28:13Z TongaSimplifier finished after 0.015 seconds
2023-12-12T17:28:13Z Running LICM
2023-12-12T17:28:14Z LICM finished after 0.006 seconds
2023-12-12T17:28:14Z Running RewriteReplicationMatmul
2023-12-12T17:28:14Z RewriteReplicationMatmul finished after 0.018 seconds
2023-12-12T17:28:14Z Running FlattenMacroLoop
2023-12-12T17:28:14Z FlattenMacroLoop finished after 0.027 seconds
2023-12-12T17:28:14Z Running SimplifyMacroPredicates
2023-12-12T17:28:14Z SimplifyMacroPredicates finished after 0.027 seconds
2023-12-12T17:28:14Z Running DataLocalityOpt
2023-12-12T17:28:14Z DataLocalityOpt finished after 0.056 seconds
2023-12-12T17:28:14Z Running TongaSimplifier
2023-12-12T17:28:15Z TongaSimplifier finished after 0.049 seconds
2023-12-12T17:28:15Z Running LegalizeSundaMacro
2023-12-12T17:28:15Z LegalizeSundaMacro finished after 0.048 seconds
2023-12-12T17:28:15Z Running TongaSimplifier
2023-12-12T17:28:15Z TongaSimplifier finished after 0.048 seconds
2023-12-12T17:28:15Z Running PerfectLoopNest
2023-12-12T17:28:16Z PerfectLoopNest finished after 0.004 seconds
2023-12-12T17:28:16Z Running FlattenMacroLoop
2023-12-12T17:28:16Z FlattenMacroLoop finished after 0.007 seconds
2023-12-12T17:28:16Z Running RewriteWeights
2023-12-12T17:28:16Z RewriteWeights finished after 0.021 seconds
2023-12-12T17:28:16Z Running ReshapeWeights
2023-12-12T17:28:16Z ReshapeWeights finished after 0.002 seconds
2023-12-12T17:28:16Z Running FlattenMacroLoop
2023-12-12T17:28:16Z FlattenMacroLoop finished after 0.004 seconds
2023-12-12T17:28:16Z Running SimplifyMacroPredicates
2023-12-12T17:28:17Z SimplifyMacroPredicates finished after 0.037 seconds
2023-12-12T17:28:17Z Running InferInitValue
2023-12-12T17:28:17Z InferInitValue finished after 0.167 seconds
2023-12-12T17:28:17Z Running TongaSimplifier
2023-12-12T17:28:17Z TongaSimplifier finished after 0.027 seconds
2023-12-12T17:28:17Z Running SimplifyTensor
2023-12-12T17:28:17Z SimplifyTensor finished after 0.015 seconds
2023-12-12T17:28:17Z Running LICM
2023-12-12T17:28:17Z LICM finished after 0.007 seconds
2023-12-12T17:28:17Z Running SundaISel
2023-12-12T17:28:18Z SundaISel finished after 0.123 seconds
2023-12-12T17:28:18Z Running LowerThorKernels
2023-12-12T17:28:18Z LowerThorKernels finished after 0.003 seconds
2023-12-12T17:28:18Z Running TongaLoopInterchange
2023-12-12T17:28:18Z TongaLoopInterchange finished after 0.003 seconds
2023-12-12T17:28:18Z Running TongaSimplifyPredicates
2023-12-12T17:28:18Z TongaSimplifyPredicates finished after 0.017 seconds
2023-12-12T17:28:18Z Running TongaLoopFusion
2023-12-12T17:28:18Z TongaLoopFusion finished after 0.013 seconds
2023-12-12T17:28:18Z Running TongaLoopInterchange
2023-12-12T17:28:19Z TongaLoopInterchange finished after 0.017 seconds
2023-12-12T17:28:19Z Running TongaLICM
2023-12-12T17:28:19Z TongaLICM finished after 0.016 seconds
2023-12-12T17:28:19Z Running FactorizeBlkDims
2023-12-12T17:28:19Z FactorizeBlkDims finished after 0.008 seconds
2023-12-12T17:28:19Z Running TongaInstComb
2023-12-12T17:28:19Z TongaInstComb finished after 0.067 seconds
2023-12-12T17:28:19Z Running TongaValueNumbering
2023-12-12T17:28:19Z TongaValueNumbering finished after 0.005 seconds
2023-12-12T17:28:19Z Running TongaInstComb
2023-12-12T17:28:20Z TongaInstComb finished after 0.019 seconds
2023-12-12T17:28:20Z Running VectorizeDMA
2023-12-12T17:28:20Z VectorizeDMA finished after 0.002 seconds
2023-12-12T17:28:20Z Running TongaSimplifyPredicates
2023-12-12T17:28:20Z TongaSimplifyPredicates finished after 0.003 seconds
2023-12-12T17:28:20Z Running LegalizePartitionReduce
2023-12-12T17:28:20Z LegalizePartitionReduce finished after 0.006 seconds
2023-12-12T17:28:20Z Running DeConcat
2023-12-12T17:28:20Z DeConcat finished after 0.025 seconds
2023-12-12T17:28:20Z Running PartialSimdFusion
2023-12-12T17:28:20Z PartialSimdFusion finished after 0.003 seconds
2023-12-12T17:28:20Z Running TritiumFusion
2023-12-12T17:28:21Z TritiumFusion finished after 0.002 seconds
2023-12-12T17:28:21Z Running CCOpFusion
2023-12-12T17:28:21Z CCOpFusion finished after 0.006 seconds
2023-12-12T17:28:21Z Running VectorizeMatMult
2023-12-12T17:28:21Z VectorizeMatMult finished after 0.033 seconds
2023-12-12T17:28:21Z Running PartialLoopFusion
2023-12-12T17:28:21Z PartialLoopFusion finished after 0.004 seconds
2023-12-12T17:28:21Z Running TongaLICM
2023-12-12T17:28:23Z TongaLICM finished after 0.042 seconds
2023-12-12T17:28:23Z Running LowerTranspose
2023-12-12T17:28:23Z LowerTranspose finished after 0.034 seconds
2023-12-12T17:28:23Z Running LateTongaInstComb
2023-12-12T17:28:23Z LateTongaInstComb finished after 0.042 seconds
2023-12-12T17:28:23Z Running SplitAccGrp
2023-12-12T17:28:23Z SplitAccGrp finished after 0.009 seconds
2023-12-12T17:28:23Z Running SpillPSum
2023-12-12T17:28:23Z SpillPSum finished after 0.040 seconds
2023-12-12T17:28:23Z Running LowerIntrinsics
2023-12-12T17:28:23Z LowerIntrinsics finished after 0.010 seconds
2023-12-12T17:28:23Z Running LegalizeType
2023-12-12T17:28:23Z LegalizeType finished after 0.011 seconds
2023-12-12T17:28:23Z Running TongaLICM
2023-12-12T17:28:23Z TongaLICM finished after 0.040 seconds
2023-12-12T17:28:23Z Running InferPSumTensor
2023-12-12T17:28:23Z InferPSumTensor finished after 0.039 seconds
2023-12-12T17:28:23Z Running WeightCoalescing
2023-12-12T17:28:23Z WeightCoalescing finished after 0.010 seconds
2023-12-12T17:28:23Z Running LegalizeSundaAccess
2023-12-12T17:28:23Z LegalizeSundaAccess finished after 0.041 seconds
2023-12-12T17:28:23Z Running TernaryFission
2023-12-12T17:28:23Z TernaryFission finished after 0.058 seconds
2023-12-12T17:28:23Z Running RelaxPredicates
2023-12-12T17:28:23Z RelaxPredicates finished after 0.012 seconds
2023-12-12T17:28:23Z Running TensorInitialization
2023-12-12T17:28:23Z TensorInitialization finished after 0.025 seconds
2023-12-12T17:28:23Z Running TongaSimplifyPredicates
2023-12-12T17:28:24Z TongaSimplifyPredicates finished after 0.015 seconds
2023-12-12T17:28:24Z Running ExpandISAMacro
2023-12-12T17:28:24Z ExpandISAMacro finished after 0.008 seconds
2023-12-12T17:28:24Z Running SimplifyTongaTensor
2023-12-12T17:28:24Z SimplifyTongaTensor finished after 0.013 seconds
2023-12-12T17:28:24Z Running DMALocalityOpt
2023-12-12T17:28:24Z DMALocalityOpt finished after 0.002 seconds
2023-12-12T17:28:24Z Running DataStreaming
2023-12-12T17:28:24Z DataStreaming finished after 0.007 seconds
2023-12-12T17:28:24Z Running SFKVectorizer
2023-12-12T17:28:24Z SFKVectorizer finished after 0.085 seconds
2023-12-12T17:28:24Z Running LateLegalizeInst
2023-12-12T17:28:24Z LateLegalizeInst finished after 0.011 seconds
2023-12-12T17:28:24Z Running CoalesceCCOp
2023-12-12T17:28:24Z CoalesceCCOp finished after 0.005 seconds
2023-12-12T17:28:24Z Running SimpleAllReduceTiling
2023-12-12T17:28:24Z SimpleAllReduceTiling finished after 0.005 seconds
2023-12-12T17:28:24Z Running StaticProfiler
2023-12-12T17:28:24Z StaticProfiler finished after 0.019 seconds
2023-12-12T17:28:24Z Running SplitAPUnionSets
2023-12-12T17:28:24Z SplitAPUnionSets finished after 0.022 seconds
2023-12-12T17:28:24Z Running DumpGraphAndMetadata
2023-12-12T17:28:24Z DumpGraphAndMetadata finished after 0.008 seconds
2023-12-12T17:28:24Z Running BirCodeGenLoop
2023-12-12T17:28:24Z BirCodeGenLoop finished after 0.073 seconds
2023-12-12T17:28:24Z Running birverifier
2023-12-12T17:28:24Z birverifier finished after 0.000 seconds
2023-12-12T17:28:24Z Running expand_replication
2023-12-12T17:28:24Z expand_replication finished after 0.000 seconds
2023-12-12T17:28:24Z Running unroll
2023-12-12T17:28:24Z unroll finished after 0.005 seconds
2023-12-12T17:28:24Z Running psum_legalization
2023-12-12T17:28:24Z psum_legalization finished after 0.000 seconds
2023-12-12T17:28:24Z Running pre_opts
2023-12-12T17:28:24Z pre_opts finished after 0.000 seconds
2023-12-12T17:28:24Z Running error_injector
2023-12-12T17:28:24Z error_injector finished after 0.000 seconds
2023-12-12T17:28:24Z Running constant_propagate
2023-12-12T17:28:24Z constant_propagate finished after 0.003 seconds
2023-12-12T17:28:24Z Running vn_splitter
2023-12-12T17:28:24Z vn_splitter finished after 0.000 seconds
2023-12-12T17:28:24Z Running lower_ac
2023-12-12T17:28:24Z lower_ac finished after 0.000 seconds
2023-12-12T17:28:24Z Running input_dma_coalescing
2023-12-12T17:28:24Z input_dma_coalescing finished after 0.000 seconds
2023-12-12T17:28:24Z Running pre_sched
2023-12-12T17:28:24Z pre_sched finished after 0.003 seconds
2023-12-12T17:28:24Z Running mm_packing
2023-12-12T17:28:24Z mm_packing finished after 0.003 seconds
2023-12-12T17:28:24Z Running coloring_allocator_psum
2023-12-12T17:28:24Z coloring_allocator_psum finished after 0.000 seconds
2023-12-12T17:28:24Z Running dma_optimization_psum
2023-12-12T17:28:24Z dma_optimization_psum finished after 0.000 seconds
2023-12-12T17:28:24Z Running address_rotation_psum
2023-12-12T17:28:24Z address_rotation_psum finished after 0.000 seconds
2023-12-12T17:28:24Z Running coloring_allocator_sb
2023-12-12T17:28:24Z coloring_allocator_sb finished after 0.002 seconds
2023-12-12T17:28:24Z Running address_rotation_sb
2023-12-12T17:28:24Z address_rotation_sb finished after 0.002 seconds
2023-12-12T17:28:24Z Running dma_optimization_sb
2023-12-12T17:28:24Z dma_optimization_sb finished after 0.002 seconds
2023-12-12T17:28:24Z Running address_rotation_sb
2023-12-12T17:28:24Z address_rotation_sb finished after 0.001 seconds
2023-12-12T17:28:24Z Running coloring_allocator_dram
2023-12-12T17:28:24Z coloring_allocator_dram finished after 0.000 seconds
2023-12-12T17:28:24Z Running address_rotation_dram
2023-12-12T17:28:24Z address_rotation_dram finished after 0.000 seconds
2023-12-12T17:28:24Z Running tensorcopy_accel
2023-12-12T17:28:24Z tensorcopy_accel finished after 0.000 seconds
2023-12-12T17:28:24Z Running peephole_opts
2023-12-12T17:28:24Z peephole_opts finished after 0.000 seconds
2023-12-12T17:28:24Z Running lower_kernel
2023-12-12T17:28:24Z lower_kernel finished after 0.000 seconds
2023-12-12T17:28:24Z Running build_fdeps
2023-12-12T17:28:24Z build_fdeps finished after 0.001 seconds
2023-12-12T17:28:24Z Running remove_redundancies
2023-12-12T17:28:24Z remove_redundancies finished after 0.000 seconds
2023-12-12T17:28:24Z Running anti_dependency_analyzer
2023-12-12T17:28:24Z anti_dependency_analyzer finished after 0.008 seconds
2023-12-12T17:28:24Z Running tensor_copy_elim
2023-12-12T17:28:24Z tensor_copy_elim finished after 0.000 seconds
2023-12-12T17:28:24Z Running post_sched
2023-12-12T17:28:24Z post_sched finished after 0.002 seconds
2023-12-12T17:28:24Z Running address_rotation_sb
2023-12-12T17:28:24Z address_rotation_sb finished after 0.002 seconds
2023-12-12T17:28:24Z Running anti_dependency_analyzer
2023-12-12T17:28:24Z anti_dependency_analyzer finished after 0.002 seconds
2023-12-12T17:28:24Z Running dep_opt
2023-12-12T17:28:24Z dep_opt finished after 0.010 seconds
2023-12-12T17:28:24Z Running report_stats
2023-12-12T17:28:24Z report_stats finished after 0.000 seconds
2023-12-12T17:28:24Z Running assign_trigger_engine
2023-12-12T17:28:24Z assign_trigger_engine finished after 0.000 seconds
2023-12-12T17:28:24Z Running alloc_queues
2023-12-12T17:28:24Z alloc_queues finished after 0.000 seconds
2023-12-12T17:28:24Z Running dep_reduction
2023-12-12T17:28:24Z dep_reduction finished after 0.001 seconds
2023-12-12T17:28:24Z Running bir_racecheck
2023-12-12T17:28:24Z bir_racecheck finished after 0.002 seconds
2023-12-12T17:28:24Z Running lower_dma
2023-12-12T17:28:24Z lower_dma finished after 0.000 seconds
2023-12-12T17:28:24Z Running coalesce_dma_blocks
2023-12-12T17:28:24Z coalesce_dma_blocks finished after 0.000 seconds
2023-12-12T17:28:24Z Running alloc_semaphores
2023-12-12T17:28:24Z alloc_semaphores finished after 0.000 seconds
2023-12-12T17:28:24Z Running expand_inst_late
2023-12-12T17:28:24Z expand_inst_late finished after 0.000 seconds
2023-12-12T17:28:24Z Running lower_sync
2023-12-12T17:28:24Z lower_sync finished after 0.000 seconds
2023-12-12T17:28:24Z Running lower_act
2023-12-12T17:28:24Z lower_act finished after 0.001 seconds
2023-12-12T17:28:24Z Running lower_dve
2023-12-12T17:28:24Z lower_dve finished after 0.002 seconds
2023-12-12T17:28:24Z Running lower_ap
2023-12-12T17:28:24Z lower_ap finished after 0.000 seconds
2023-12-12T17:28:24Z Running alloc_regs
2023-12-12T17:28:24Z alloc_regs finished after 0.000 seconds
2023-12-12T17:28:24Z Running birverifier
2023-12-12T17:28:24Z birverifier failed after 0.000 seconds
2023-12-12T17:28:24Z
2023-12-12T17:28:24Z Diagnostic information:
2023-12-12T17:28:24Z   NeuronX Compiler version 2.12.46.0+8a56b1c94
2023-12-12T17:28:24Z
2023-12-12T17:28:24Z   Python version 3.8.10
2023-12-12T17:28:24Z   HWM version 2.12.0.0-422c9037c
2023-12-12T17:28:24Z   NumPy version 1.24.4
2023-12-12T17:28:24Z
2023-12-12T17:28:24Z   Running on AMI ami-01257e71ecb2f431c
2023-12-12T17:28:24Z   Running in region usw2-az4
2023-12-12T17:28:24Z
2023-12-12T17:28:24Z Diagnostic logs stored in /home/ubuntu/dev/Kholinar/xla/log-neuron-cc.txt

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Using:
```
$ dpkg-query -W -f='${binary:Package} ${Version}\n' | grep '^aws-neuron'
aws-neuronx-collectives 2.19.5.0-530fb3064
aws-neuronx-dkms 2.15.9.0
aws-neuronx-gpsimd-customop-lib 0.9.1.0
aws-neuronx-gpsimd-tools 0.9.0.0-e7d693355
aws-neuronx-runtime-lib 2.19.4.0-ed642d216
aws-neuronx-tools 2.16.1.0
$ neuronx-cc --version
NeuronX Compiler version 2.12.46.0+8a56b1c94

Python version 3.8.10
HWM version 2.12.0.0-422c9037c
NumPy version 1.24.4

Running on AMI ami-01257e71ecb2f431c
Running in region usw2-az4
```

**Comment by jeffhataws** [comment]
Thanks @xanderdunn . We have reproduced and fixed the problem. The fix will be available in an upcoming release.

**Comment by xanderdunn** [comment]
These are fixed on `neuronx_cc-2.12.54.0+f631c2365` Thank you! :)
I'll close this when it reaches release.

**Comment by xanderdunn** [comment]
Fixed in 2.16 release
